Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -384,6 +395,27 @@ protected override async Task OnAfterRenderAsync( bool firstRender )
         await base.OnAfterRenderAsync( firstRender );
     }
 
+    private void AutoGenerateColumns()
+    {
+        var properties = ReflectionHelper.GetPublicProperties<TItem>();
+        foreach ( var property in properties )
+        {
+            if ( !( property.PropertyType.IsValueType || property.PropertyType == typeof( string ) ) )
+            {
+                continue;
+            }
+
+            var column = new DataGridColumn<TItem>
+            {
+                Caption = ReflectionHelper.ResolveCaption( property ),
+                Field = property.Name,
+            };
+            this.AddColumn( column );
+        }
+
+        StateHasChanged();

```

Comment:
`InvokeAsync(StateHasChanged)`

Reply:
It's redudant the thread is synced up here. Specially since it isn't async.